### PR TITLE
Setup ecs configs for worker process

### PR DESCRIPTION
**Pull request by fuziontech**
## Changes

This reworks how we are going to be deploying to AWS ECS services on the `posthog-production` cluster.

There are three services outlined in task defintions:
- posthog-production-web
  services:
    - web
    - event
- posthog-production-worker
- posthog-production-migration (not a service, just a task)

Currently these are all based off of the same Posthog prod docker image built from the dockerfile found in posthog-production repo.

It's setup so that the services can be swapped out relatively easily without new configs, only modifications to what exists here. This will make migrating the event service to golang later relatively easy.

The github action is setup so that on landing to master
- We build the production docker image and store it into Elastic Container Registry
- Create new task definition revisions with the SHA for those images
- Instruct the ECS Fargate cluster to run the migration task
- Update service definitions for web, worker, and event services which causes them to do a rolling update
- Health checks are made before the ELB target groups are switched over to point to the new revision of the service
- Success!

### CODE REVIEW

File: task-definition.worker.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                    "valueFrom": "arn:aws:secretsmanager:us-east-1:795637471508:secret:Posthog_Production_Heroku-FQ2itU:STRIPE_WEBHOOK_SECRET::"
+                }
+            ],
+            "entryPoint": ["sh", "-c"],
+            "command": ["./bin/docker-worker"]
+        }
+    ],
+    "requiresCompatibilities": ["FARGATE"],
+    "cpu": "512",
+    "memory": "1024"
```
Comment (EDsCODE) [inline task-definition.worker.json:204]: don't know much about this but should these values be the same across all definitions

File: ee/kafka_client/client.py [hunk 2]
```diff
@@ -8,8 +8,9 @@
 from kafka import KafkaProducer as KP
 
 from ee.clickhouse.client import async_execute, sync_execute
+from ee.kafka_client import helper
```
Comment (EDsCODE) [inline ee/kafka_client/client.py:11]: Why was this needed to be moved into our repo rather than using the package

**Review by EDsCODE — COMMENTED** [review]
Not in this PR but I was just looking around and catching up on how posthog.net is deployed and I think it's really amusing how we've re-hooked `posthog-prod` which should be an auxiliary repo dependent on `posthog` to be retrieved in this repo and autodeployed . Now all we need to do his have a workflow on `posthog-prod` that auto deploys and then both repos can have a deploy race. The most continuous of deployments

File: ee/kafka_client/client.py (hunk 2 above)
Comment (fuziontech) [reply ee/kafka_client/client.py:11]: because of base64 encoded keys. Going to move everything to the vendored bits, but that'll be a cleanup task.

**Comment by fuziontech** [comment]
> Not in this PR but I was just looking around and catching up on how posthog.net is deployed and I think it's really amusing how we've re-hooked `posthog-prod` which should be an auxiliary repo dependent on `posthog` to be retrieved in this repo and autodeployed . Now all we need to do his have a workflow on `posthog-prod` that auto deploys and then both repos can have a deploy race. The most continuous of deployments

We take Continuous Deployment seriously here at Posthog. In fact we take it so much to heart that we are Continuously Deploying ™️ . We are never not deploying. You are served fresh bits no matter how old the change is. We have found this to improve user experience and reduce defects commonly found in stale code.

File: task-definition.worker.json (hunk 1 above)
Comment (fuziontech) [reply task-definition.worker.json:204]: To start out with that's the idea. That will probably change later as we watch the metrics.

File: ee/kafka_client/client.py (hunk 2 above)
Comment (EDsCODE) [reply ee/kafka_client/client.py:11]: oh got it

**Review by EDsCODE — APPROVED** [review]
Not the most familiar with ecs task configs/how the roles/groups are config'd in aws but the setup mirrors the first task def and the environments are defined as expected
